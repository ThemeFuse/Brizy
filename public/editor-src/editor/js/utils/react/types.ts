import { FC, PropsWithChildren } from "react";

// Custom Type for a React functional component with props AND CHILDREN
export type FCC<P = NonNullable<unknown>> = FC<PropsWithChildren<P>>;

export type FCP<P, R> = (p: P) => R;
