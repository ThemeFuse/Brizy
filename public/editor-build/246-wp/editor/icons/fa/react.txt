JBH2TsV4vQrowt0cuBL0wXipF3x3xs53Gs5pwgwpGdUqGW9txgwcCBTjTRxWv3a9CdUaGWU1GNCaHNYsCd48wAZ0uWVeJMDHHXY4FdCaGNw3FdDdFNOoHW0rFdanGNUoIW0tFdOnGNSoGc01FdYoIM0tFdwaGM43FNwoHWUsFdOnGNYoGMUrGc4tFNO5FdSaHW4sFNYqHs41FNCtFdYnGNCtFdGnGdSoGs0rHM4rFNS5FdCoHc0rGNCoHcUtIW40FNKoGsUtFdwnIW41CXwoHc0rGc41CXYrFdOnGc43FNCoHc01FdOnHM4sFNaoGs03FdwnHXOoHM00GW40FNerFdYnHNwoHW0rGNaoHW00GM41FNC2FdCaGNOoGc0tHWU2GW4tFNCtCXYrHc43CXYoGMU1FdSaGc4tCXYrFdYaGs43CXY2FdwnHc40CXYoIW0rGc43CXGoIW0rIW42CXOoIOGtIW4tCXY5Hc4sCXUaGdC1FdKaGWUsHNOoHgGqCXGrFdCaHXUoIWU2Gc41CXe2FdGaIXYoHMU0FdOaGM41CXeaGsUrGs42CXKoGs0rFdOaHc0sFdaaGNYoIM00CXY4FNYqFdOaHNOoHM0sFdGaINeoHMUsGs45CXYrHW42CXC3CXY1FdSaHtCoHW0oHWUrGNSoHc0tIM4rCXGoHM0tFdYaHs02FdGaGNUoHM05FdwaHW40CXKoGsU5CXaoHWUrGs42CXYsFdKaHXCoIWUtHc44CXa1FdYaHNYoHsUrGNYoGcUtHc42CXC3FNY1FdSaGtOoIW02Gc45CXC0FdKnGNCqFdOnFdenHW40FNYoIM04FdenGs0rGs41CXGoGc0oIMU2FdGnGM45CXeoHW0sFdeaHNwoHs0rIM4rCXe5FdOnHNUaINeoHM04GM43CXUnGtUoGs0tIM40FNO5FdwnINGoIW03IW40ye0sIXCoIMU5Gc4tStG3FdCnGtCoHWU3GM45FNK1FdYaIXwoHs0tHcUrHc45CXeoHsUsGs40CXK4FdeaGNCoIWUrGXUoHW0oHsUtFdKnGM40CXSoHs0sFdGaGNUnGdCoGc01FNK0FdwnIW42FNS3FdGnGNUoHc0rGs0rIW42FNC3FdCnGtSoHW00Gc42FNOtFdYaGs45FNGoHsU3FdwnHs4sCXYrFdwnGNUoH3jHGNS3FdCaGtU3FdPdHM4rCXaoHsUrGW4tCXY3FdKaGNOoIWUsHM45FNY1FdSnGM43FNGrFdYnHW4sFNK2FdKnHs41CXKoHW0rHW40CXeoIM0sIM4tCXY2FdGnHXKoHMU0FdSaIW44CXeoGsUrHs41CXY0FdGaGdSoGRjnFNGqFdGnGNCqFdHdGNKoHW0tFdCaGdeoHs01FdaaHXOoHc03FdanHM4tCXaoGs0rGW41CXY2FdanGNOoHWUsHM40FNKoIMU4FdOnIM43CXY3FdCnGNKoGcUsHc02FdGnGNKoIM0rGM42FNC5FdOnGNSnHXGoHhjnGdwoHWU2IW45StSoHc0rGs44CXYtFdanGdwoGsUsGM40FNKqFdTtGNOoIW0sHc4sCXC0FdKnGtaoIQGrHM0rFdYaGtUoGs0rFdwaHXOoIM0rFdxtGtYaFdSaHXOoIMUrFdxdIW41CXYsFdSaGNSoHcUsHM41CXC0FdGaGtaoH3GrHW45CXC2FdwaGdYoHsU0GW40Ss02FdwaGNGoIW0rGs45CXC3FdKnGdYoHcU0GW44FNwoHcUrGs4tFNY1FdwaGdSoGc0sHW4sCXG5FNY0FdeaGM4rFNGqFdKaGM42FNK2FdYaGM42ws0tGW45FM41FNK1FdSnGM40Ss04FdwnGNCoHs0rHc45FNC1FdwnGdKoHc0tIRGnGNKoIW0sHc44FNCrFdOnHXUoHhjnGNaqFdSaHNYoGgG1FdYnIW44CXeoIM0rHs43GW43HXKqGterGXKqHdw4GXG0CXY0FdSnGdSoHsU2FdKaGNKoHMUrGcUsIM4sCXY2FdeaHXKoGs0rHM41CXGoHM0tGM4sCXSoGc00HsU4CXOoHW04FdKaGNUoHM0rHsUrHM41FNC1FdT6vNY0FdKnHtSoHQGnHW43FNaoIW05FdOnGNwoHc0rHW41FNC2FdCnHW45FNaoHM0rGW0rHc45FNY1FdGnGdOoGcUrHc4rCXCaGtYoHMU0FdwaHXOoIMU4FNKoHcUrHW44FNYqCXC5FdCnGNSoGMU0Gs40ye0sHNSoGcUrGNaoG2GrGW41CXYrFdKaGdUoHWUsGs40CXC5FdSaGtOoIW0rIM44FM45FNG5FdwnFdenHNeoHMUqCXeoIW0rGc45CXY5FdenGdKoIMUsIM45FNG1Fdb6NNY0GW4sCXO3StY2FdanIM44CXO0FdYaHW4sCXetFdKaGteaGc41CXCoGcU1CXKoHcU3FdSaHs0rHM41CXY2FdwnGdeoIWUtHW41FNKsFdeaHNGoGM0sGc42CXCnHXOaHM41FNS3FdCaGNUoHW0rFdGnHM4rFNCoHW0rGW4tFNGoHM0rHM41FNeoHW00IW40FNGoGc04HW45CXYsFdSnINL6vM0sHW41CXC2Gs42Ss00FdCnGM4sFNaoGs0sFdOnGNCoHW0tFdenGdYoGs02FdwnHXOoHM0rHs4tFNStFNGrFdCnGNUoGM03FNY2FdenGNwoIW0rIW44FNC5FdeaGW0rIW4tCXGrFdSnHXYoHsU3Hs4sFNO3FdSaHM43FNCaGNYoHM0tFdaaGNwoGs01FdOaHc44CXCrFdwaGNOaHXGaGdKoHMU2Gs42FNeoHcUsGW45FNY3FdeaHXCoHM0sHW44CXS0FdP6vNYrHc42CXe4Ss0rHc41CXY1FdYnGtOoHcUsHs4rFNO2FdKaGtOoGs0rGM4rCXOoGs0sGs45CXOoIW0tHM4tCXYoGs0rHM45FNeoGc0sGc41FNK0FdOnGNGoHM05GcUrFdYnHM42CXCoGs0rGM4sCXGoHs0rHc43CXCsFdKaHW44CXK1CXaoGMU2Hs45CXeoIWUrGs4sCXY4FdwaGdwoHsUtHc42CXKtFdCaHNGoHW0tFdCaGs4rFNSoHWU2FdYnIM42CXaoIRjnGdKoHM0sHW4tSs0rGW4sFNYrFNCqFdKnGdGoGc0tGW4tFNG2FdGaIM42FdKaGNeoHM42CXC5FdOoHcUrGW4tCXUaGdUoHW0oGcUtGW40FM43FNeoGcUrGc43FNY5FdYaGdKoIW0sIM42CXG2FdL6vNYtGW43CXGqSs0oIMUrGc4sFNSoIMUsGs42FNY2FdOaGtYoGs0rHM45CXeoGc00IM44FNCoIW04Hc40FNG0FdCnHW4sFNGoHc04FdKnHs41FNYsFdwnGNYoHMUrHM4tFNY2FdeaGdeoHW0tHW44CXKsFdCnHNGoHcUsGc45FNYoIMU0HM43FNOoHWU2IW4sFNYqFdOaGMU0FdYaGM45CXaoGcUsFdwaGNCoGcU0FdeaGdYoHcU1FdwaHXKoGMUsFdOaHdSoG3jnGNaoGc0rGXwoHQGnGc44FdenHM42CXYoIW04FdOaGc42FNwnGdYoIW0rHM42FNKtFdYnGdOoHM02Gs44CXeoHc0sGW40CXY3FdwnHXYoHWUsHW41FNSsFdeaHM4sCXYoHMUrGW4sCXGoGMUrHMU0FdwaHXSoHcUrHcU3IM4tCXG5FdaaHteoGsU1IWUqCXY5FdSnGtKoIMU0HW45FNa0FdaaHdYoHBjnFNY0IM43FNY1StC1FdGaGWU0HM44FNCqFdOaHXOoIW00HM44ws0sGW41FNK1FdanHXOoIW00HM44Ss0sHM4tCXUnHXOoIWUsGW41FNK1FdaaHXOoIBGsGW41CXK1FdaaHXOoIWU0HM44ycCpJdqpw3ThJa==