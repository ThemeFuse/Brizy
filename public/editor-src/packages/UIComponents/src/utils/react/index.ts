import { SyntheticEvent } from "react";

type WithValue = HTMLElement & { value: string };

export const inputValue = (e: SyntheticEvent<WithValue>): string =>
  (e.target as WithValue).value;
