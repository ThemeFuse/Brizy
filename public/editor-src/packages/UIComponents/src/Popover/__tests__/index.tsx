import { fireEvent, render, waitFor } from "@testing-library/react";
import React, { ReactNode } from "react";
import { Popover } from "../index";

type Props = {
  onOpen: () => void;
  onClose: () => void;
  clickOutsideExceptions: string[];
  children: ReactNode;
  className: string;
};

const renderPopover = ({
  onOpen,
  onClose,
  clickOutsideExceptions,
  children,
  className
}: Partial<Props>) => {
  const { container, getByText } = render(
    <>
      <Popover
        size="small"
        placement="top"
        trigger={<div className="trigger">trigger</div>}
        onOpen={onOpen}
        onClose={onClose}
        clickOutsideExceptions={clickOutsideExceptions}
        className={className}
      >
        {children}
      </Popover>
      {clickOutsideExceptions?.length && (
        <div className="exception">exception</div>
      )}
    </>
  );

  return { container, getByText };
};

describe("Test Popover", () => {
  describe("Snapshots", () => {
    test("Render popover with small tooltip and className", async () => {
      const { container, getByText } = renderPopover({
        className: "test-classname"
      });

      const trigger = getByText("trigger");
      const popover = container.querySelector(".brz-ed-control__popover");

      fireEvent.click(trigger);

      const tooltipOverlay = container.querySelector(
        ".brz-ed-tooltip__overlay"
      );

      await waitFor(() => {
        expect(tooltipOverlay).toHaveClass("brz-ed-tooltip--small");
        expect(popover).toHaveClass("test-classname");
        expect(container).toMatchSnapshot();
      });
    });

    test("Render opened popover", async () => {
      const { container, getByText } = renderPopover({
        children: "children"
      });

      const trigger = getByText("trigger");

      fireEvent.click(trigger);

      const popoverContent = getByText("children");

      await waitFor(() => {
        expect(popoverContent).toBeInTheDocument();
        expect(container).toMatchSnapshot();
      });

      fireEvent.click(trigger);

      await waitFor(() => {
        expect(popoverContent).not.toBeInTheDocument();
      });
    });
  });

  test("Check onOpen, onClose", () => {
    const onOpen = jest.fn();
    const onClose = jest.fn();

    const { getByText } = renderPopover({ onOpen, onClose });

    const trigger = getByText("trigger");

    fireEvent.click(trigger);

    expect(onOpen).toHaveBeenCalledTimes(1);
    expect(onClose).toHaveBeenCalledTimes(0);

    fireEvent.click(trigger);

    expect(onOpen).toHaveBeenCalledTimes(1);
    expect(onClose).toHaveBeenCalledTimes(1);
  });

  test("Check clickOutsideExceptions", async () => {
    const { getByText } = renderPopover({
      clickOutsideExceptions: [".exception"],
      children: "children"
    });

    const trigger = getByText("trigger");
    const exception = getByText("exception");

    fireEvent.click(trigger);

    const popoverContent = getByText("children");

    await waitFor(() => {
      expect(popoverContent).toBeInTheDocument();
    });

    fireEvent.click(exception);

    await waitFor(() => {
      expect(popoverContent).toBeInTheDocument();
    });
  });
});
