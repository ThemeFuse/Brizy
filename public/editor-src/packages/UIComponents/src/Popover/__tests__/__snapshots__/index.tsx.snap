// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Popover Snapshots Render opened popover 1`] = `
<div>
  <div
    class="brz-ed-control__popover"
  >
    <div
      class="brz-ed-tooltip brz-ed-tooltip__static brz-ed-tooltip--opened brz-ed-control__popover--tooltip"
    >
      <div
        class="brz-ed-tooltip__content"
      >
        <div
          class="trigger"
        >
          trigger
        </div>
      </div>
      <div
        class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--small"
        style="position: absolute; left: 0px; top: 0px;"
      >
        <div
          class="brz-ed-arrow brz-ed-arrow--top"
          style="position: absolute; margin: 0px;"
        />
        <div
          class="brz-ed-control__popover--content"
        >
          children
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Popover Snapshots Render popover with small tooltip and className 1`] = `
<div>
  <div
    class="brz-ed-control__popover test-classname"
  >
    <div
      class="brz-ed-tooltip brz-ed-tooltip__static brz-ed-tooltip--opened brz-ed-control__popover--tooltip"
    >
      <div
        class="brz-ed-tooltip__content"
      >
        <div
          class="trigger"
        >
          trigger
        </div>
      </div>
      <div
        class="brz-ed-animated brz-ed-animated--fadeInUp brz-ed-tooltip__overlay brz-ed-tooltip--small"
        style="position: absolute; left: 0px; top: 0px;"
      >
        <div
          class="brz-ed-arrow brz-ed-arrow--top"
          style="position: absolute; margin: 0px;"
        />
        <div
          class="brz-ed-control__popover--content"
        />
      </div>
    </div>
  </div>
</div>
`;
